tests/Functional/TestCase.php

```diff
@@ -82,16 +86,114 @@ protected static function runFrontScript(array $args, string $cwd = __DIR__): Pr
         return $process;
     }
 
-    protected static function createFiles(array $filenames): void
+    protected static function createFiles(string $baseDir, array $filenames): void
     {
         foreach ($filenames as $filename) {
-            $dirname = dirname($filename);
-            if (!file_exists($dirname)) {
-                self::assertTrue(mkdir($dirname, 0777, true), 'Created directory.');
+            self::createFile($baseDir, $filename);
+        }
+    }
+
+    protected static function createFile(string $baseDir, string $filename): void
+    {
+        $filename = "{$baseDir}/{$filename}";
+        $dirname = dirname($filename);
+        if (!file_exists($dirname)) {
+            self::assertTrue(mkdir($dirname, 0777, true), 'Created directory.');
+        }
+        self::assertTrue(touch($filename), 'Created file.');
+        self::assertNotFalse(realpath($filename), 'Got absolute path.');
+    }
+
+    protected static function changeFile($dir, $filename): void
+    {
+        file_put_contents(
+            DirectoryUtil::ensureTrailingSlash($dir) . $filename,
+            self::CHANGED_CONTENT
+        );
+    }
+
+    protected static function deleteFile($dir, $filename): void
+    {
+        unlink(DirectoryUtil::ensureTrailingSlash($dir) . $filename);
+    }
+
+    protected static function fixSeparators(string $path)
+    {
+        return str_replace(['/', '\\'], DIRECTORY_SEPARATOR, $path);
+    }
+
+    protected static function fixSeparatorsMultiple(&...$paths): void
```

Comment:
Why is $paths passed by reference?

Reply:
Because the method isn't returning anything--it's actually modifying the variables passed in. It's really just a convenience wrapper around `::fixSeparators()` in case of multiple variables. To illustrate:

```diff
- $one = self::fixSeparators($one);
- $two = self::fixSeparators($two);
- $three = self::fixSeparators($three);
- $four = self::fixSeparators($four);
- $five = self::fixSeparators($five);
+ self::fixSeparatorsMultiple($one, $two, $three, $four, $five);
```

Reply:
I consider this a bit a code smell, to be frank, but it's a protected method of a test class, so I don't feel very strongly about it.